tls/s2n_async_pkey.c

```diff
@@ -409,3 +407,102 @@ S2N_RESULT s2n_async_pkey_sign_free(struct s2n_async_pkey_op *op)
 
     return S2N_RESULT_OK;
 }
+
+int s2n_async_get_op_type(struct s2n_async_pkey_op *op, s2n_async_pkey_op_type * type)
+{
+    POSIX_ENSURE_REF(op);
+    POSIX_ENSURE_REF(type);
+
+    switch (op->type) {
+        case S2N_ASYNC_DECRYPT:
+            *type = S2N_ASYNC_DECRYPT;
+            break;
+        case S2N_ASYNC_SIGN:
+            *type = S2N_ASYNC_SIGN;
+            break;
+        default:
+            POSIX_BAIL(S2N_ERR_NOT_INITIALIZED);
+            break;
+    }
+
+    return S2N_SUCCESS;
+}
+
+int s2n_async_pkey_op_get_input_size(struct s2n_async_pkey_op *op, uint32_t * data_len )
+{
+    POSIX_ENSURE_REF(op);
+    POSIX_ENSURE_REF(data_len);
+
+    if(op->type == S2N_ASYNC_DECRYPT)
+    {
```

Comment:
How do other s2n_async_pkey_op methods handle things like this that are done differently for different op types? They don't use if/else

Reply:
Will split this into separate methods